MySQL_Functions.php

```diff
@@ -10,261 +10,400 @@
-    return MySQL::getInstance()->mysql_field_len($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-field-flags.php
-function mysql_field_flags(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_flags($result, $fieldOffset);
-}
-function  mysql_fieldflags(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_flags($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-field-name.php
-function mysql_field_name(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_name($result, $fieldOffset);
-}
-function  mysql_fieldname(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_name($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-field-type.php
-function mysql_field_type(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_type($result, $fieldOffset);
-}
-function  mysql_fieldtype(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_type($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-field-table.php
-function mysql_field_table(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_table($result, $fieldOffset);
-}
-function mysql_fieldtable(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_table($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-field-seek.php
-function mysql_field_seek(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_field_seek($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-fetch-field.php
-function mysql_fetch_field(&$result, $fieldOffset = false) {
-    return MySQL::getInstance()->mysql_fetch_field($result, $fieldOffset);
-}
-// http://www.php.net/manual/en/function.mysql-num-fields.php
-function mysql_num_fields($result) {
-    return MySQL::getInstance()->mysql_num_fields($result);
-}
-function mysql_numfields($result) {
-    return MySQL::getInstance()->mysql_num_fields($result);
-}
-// http://www.php.net/manual/en/function.mysql-num-rows.php
-function mysql_num_rows($result) {
-    return MySQL::getInstance()->mysql_num_rows($result);
-}
-function mysql_numrows($result) {
-    return MySQL::getInstance()->mysql_num_rows($result);
-}
-// http://php.net/manual/en/function.mysql-info.php
-function mysql_info($link = false) {
-    return MySQL::getInstance()->mysql_info($link);
-}
-// Close all connections
-function mysql_close_all() {
-    return MySQL::getInstance()->mysql_close_all();
-}
-// is_resource function over ride
-function is_resource_custom($resource) {
-    return MySQL::getInstance()->is_resource($resource);
-}
-// get_resource_type_custom function over ride
-function get_resource_type_custom($resource) {
-    return MySQL::getInstance()->get_resource_type($resource);
+
+if (!function_exists("mysql_connect")) {
```

Comment:
This library should only be included once. I know PHP is very relaxed when it comes to allowing multiple includes, that is why I use `require_once` once in my examples to avoid just that. If this is a must for you, I would consider creating a constant and not return if the constant is set.

Reply:
I think this is for the case when native mysql extension is already present in php, so no emulation should be applied.